src/lib/compress.c

```diff
@@ -18,190 +18,85 @@
 */
 #include "internal_libreport.h"
 
-#include <lzma.h>
-#include <lz4frame.h>
+#include <archive.h>
 
-static const uint8_t s_xz_magic[6] = { 0xFD, 0x37, 0x7A, 0x58, 0x5A, 0x00 };
-static const uint8_t s_lz4_magic[4] = { 0x04, 0x22, 0x4D, 0x18 };
-
-static bool
-is_format(const char *name, const uint8_t *header, size_t hl, const uint8_t *magic, size_t ml)
+int libreport_decompress_fd(int fdi, int fdo)
 {
-    if (hl < ml)
-    {
-        log_warning("Too short header to detect '%s' file format.", name);
-        return false;
-    }
+    int retval;
+    struct archive *archive;
+    struct archive_entry *entry;
+    int r;
 
-    return memcmp(header, magic, ml) == 0;
-}
+    retval = 0;
+    archive = archive_read_new();
 
-static int
-decompress_fd_xz(int fdi, int fdo)
-{
-    uint8_t buf_in[BUFSIZ];
-    uint8_t buf_out[BUFSIZ];
+    archive_read_support_filter_all(archive);
+    archive_read_support_format_raw(archive);
 
-    lzma_stream strm = LZMA_STREAM_INIT;
-    lzma_ret ret = lzma_stream_decoder(&strm, UINT64_MAX, 0);
-    if (ret != LZMA_OK)
+    r = archive_read_open_fd(archive, fdi, 16384);
```

Comment:
Can you please factor the block/buffer size out into a named constant? It wasn't immediately apparent to me what the number designated.